src/unittests/f32_interval.spec.ts

```diff
@@ -0,0 +1,439 @@
+
+      // Double Intervals
+      { intervals: [[0, 1], [2, 5]], expected: [0, 5]},
+      { intervals: [[2, 5], [0, 1]], expected: [0, 5]},
+      { intervals: [[0, 2], [1, 5]], expected: [0, 5]},
+      { intervals: [[0, 5], [1, 2]], expected: [0, 5]},
+      { intervals: [[kValue.f32.infinity.negative, 0], [0, kValue.f32.infinity.positive]], expected: [kValue.f32.infinity.negative, kValue.f32.infinity.positive]},
+
+      // Multiple Intervals
+      { intervals: [[0, 1], [2, 3], [4, 5]], expected: [0, 5]},
+      { intervals: [[0, 1], [4, 5], [2, 3]], expected: [0, 5]},
+      { intervals: [[0, 1], [0, 1], [0, 1]], expected: [0, 1]},
+    ]
+  )
+  .fn(t => {
+    const intervals = t.params.intervals.map(arrayToInterval);
+    const expected = arrayToInterval(t.params.expected);
+
+    const got = F32Interval.span(...intervals);
+    t.expect(
+      objectEquals(got, expected),
+      `span({${intervals}}) returned ${got}. Expected ${expected}`
+    );
+  });
+
+interface CorrectlyRoundedCase {
+  value: number;
+  expected: [number, number];
+}
+
+g.test('correctlyRoundedInterval')
+  .paramsSubcasesOnly<CorrectlyRoundedCase>(
+    // prettier-ignore
+    [
+      // Edge Cases
+      { value: kValue.f32.infinity.positive, expected: [kValue.f32.positive.max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.infinity.negative, expected: [Number.NEGATIVE_INFINITY, kValue.f32.negative.min] },
+      { value: kValue.f32.positive.max, expected: [kValue.f32.positive.max, kValue.f32.positive.max] },
+      { value: kValue.f32.negative.min, expected: [kValue.f32.negative.min, kValue.f32.negative.min] },
+      { value: kValue.f32.positive.min, expected: [kValue.f32.positive.min, kValue.f32.positive.min] },
+      { value: kValue.f32.negative.max, expected: [kValue.f32.negative.max, kValue.f32.negative.max] },
+
+      // 32-bit subnormals
+      { value: kValue.f32.subnormal.positive.min, expected: [0, kValue.f32.subnormal.positive.min] },
+      { value: kValue.f32.subnormal.positive.max, expected: [0, kValue.f32.subnormal.positive.max] },
+      { value: kValue.f32.subnormal.negative.min, expected: [kValue.f32.subnormal.negative.min, 0] },
+      { value: kValue.f32.subnormal.negative.max, expected: [kValue.f32.subnormal.negative.max, 0] },
+
+      // 64-bit subnormals
+      { value: hexToF64(0x00000000, 0x00000001), expected: [0, kValue.f32.subnormal.positive.min] },
+      { value: hexToF64(0x00000000, 0x00000002), expected: [0, kValue.f32.subnormal.positive.min] },
+      { value: hexToF64(0x800fffff, 0xffffffff), expected: [kValue.f32.subnormal.negative.max, 0] },
+      { value: hexToF64(0x800fffff, 0xfffffffe), expected: [kValue.f32.subnormal.negative.max, 0] },
+
+      // 32-bit normals
+      { value: 0, expected: [0, 0] },
+      { value: hexToF32(0x03800000), expected: [hexToF32(0x03800000), hexToF32(0x03800000)] },
+      { value: hexToF32(0x03800001), expected: [hexToF32(0x03800001), hexToF32(0x03800001)] },
```

Comment:
Is this right? Shouldn't this be `expected: [hexToF32(0x03800000), hexToF32(0x03800001)]`

If so - it seems like the testing framework could do with some tightening up.

Reply:
Ah, it's an interval. Gotcha, sorry.